api/src/task/certificates/setup_cert_user_task.py

```diff
@@ -0,0 +1,196 @@
+)
+from src.task.ecs_background_task import ecs_background_task
+from src.task.task import Task
+from src.task.task_blueprint import task_blueprint
+
+logger = logging.getLogger(__name__)
+
+FUTURE_DATE = datetime_util.make_timezone_aware(datetime(2050, 1, 1), "US/Eastern").date()
+
+
+class SetupCertUserTaskStatus(StrEnum):
+    INVALID_ROLE_IDS = "Invalid role ids"
+    AGENCY_NOT_FOUND = "Agency not found"
+    LEGACY_CERTIFICATE_ALREADY_EXISTS = "LegacyCertificate already exists"
+    SUCCESS = "Success"
+    TCERTIFICATE_IS_EXPIRED = "Tcertificate is expired"
+    TCERTIFICATE_NOT_FOUND = "Tcertificate not found"
+    TCERTIFICATE_IS_MISSING_SERIAL_NUMBER = "Tcertificate is missing serial number"
+
+
+@task_blueprint.cli.command("setup-cert-user", help="Setup the LegacyCertificate and User")
+@click.option("--tcertficates-id", "-t", help="tcertificates_id on Staging Tcertificate")
+@click.option("--role-ids", "-t", help="role_id of role that needs to be added", multiple=True)
+@flask_db.with_db_session()
+@ecs_background_task(task_name="setup-cert-user")
+def setup_cert_user(db_session: db.Session, cert_id: str, role_ids: list[str]) -> None:
+    SetupCertUserTask(db_session, cert_id, role_ids).run_task()
+
+
+class SetupCertUserTask(Task):
+
+    def __init__(self, db_session: db.Session, tcertificates_id: str, role_ids: list[str]):
+        super().__init__(db_session)
+        self.tcertificates_id = tcertificates_id
+        self.role_ids = role_ids
+        self.valid_expiration_date = FUTURE_DATE
+
+    def run_task(self) -> None:
+        with self.db_session.begin():
+            self.setup_cert()
+
+    def setup_cert(self) -> SetupCertUserTaskStatus:
+        logger.info("setup cert user start")
+        roles = self.get_roles()
+        if roles is None:
+            return SetupCertUserTaskStatus.INVALID_ROLE_IDS
+
+        tcertificate = self.get_tcertificate()
+        if tcertificate is None:
+            logger.warning("Tcertificate not found")
+            return SetupCertUserTaskStatus.TCERTIFICATE_NOT_FOUND
+        if not tcertificate.serial_num:
+            logger.warning("Tcertificate is missing serial number")
+            return SetupCertUserTaskStatus.TCERTIFICATE_IS_MISSING_SERIAL_NUMBER
+        self.valid_expiration_date = tcertificate.expirationdate or FUTURE_DATE
+        if self.valid_expiration_date <= datetime_util.get_now_us_eastern_date():
+            logger.warning("Cert is expired")
+            return SetupCertUserTaskStatus.TCERTIFICATE_IS_EXPIRED
+        if self.is_existing_certificate(tcertificate):
+            logger.warning("LegacyCertificate already exists")
+            return SetupCertUserTaskStatus.LEGACY_CERTIFICATE_ALREADY_EXISTS
+
+        agency, related_agencies = self.get_agencies(tcertificate)
+        if agency is None:
+            return SetupCertUserTaskStatus.AGENCY_NOT_FOUND
+
+        else:
+            self.process_cert_user(roles, tcertificate, agency, related_agencies)
```

Comment:
Don't store the date on the class if we calculate it in the function that calls the one place it uses it. Just pass that into that function.

```suggestion
        valid_expiration_date = tcertificate.expirationdate or FUTURE_DATE
        if valid_expiration_date <= datetime_util.get_now_us_eastern_date():
            logger.warning("Cert is expired")
            return SetupCertUserTaskStatus.TCERTIFICATE_IS_EXPIRED
        if self.is_existing_certificate(tcertificate):
            logger.warning("LegacyCertificate already exists")
            return SetupCertUserTaskStatus.LEGACY_CERTIFICATE_ALREADY_EXISTS

        agency, related_agencies = self.get_agencies(tcertificate)
        if agency is None:
            return SetupCertUserTaskStatus.AGENCY_NOT_FOUND

        else:
            self.process_cert_user(roles, tcertificate, agency, related_agencies, valid_expiration_date)
```